File: TaskManagementAPI/Entities/Models/ProjectMember.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Entities.Models
{
    public class ProjectMember
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long ProjectMemberSequence { get; set; } 
        public long ProjectId { get; set; }
        public string? AccountId { get; set; }
        public ProjectMemberRole Role { get; set; } = ProjectMemberRole.Member;
        public DateTime JoinedAt { get; set; } = DateTime.UtcNow;
        public DateTime? LeftAt { get; set; }
        [NotMapped]
        public bool IsActive => !LeftAt.HasValue;
        public Project? Project { get; set; }
        public Account? Account { get; set; }
    }

    public enum ProjectMemberRole
    {
        Owner = 0,
        Manager = 1,
        Member = 2
    }
}

File: TaskManagementWeb/src/types/projectMember.ts
export default interface ProjectMember {
    id: string;
    accountId: string;
    role: ProjectMemberRole;
    joinedAt: string;
    leftAt?: string;
    isActive: boolean;
    accountFirstName: string;
    accountLastName: string;
    accountEmail: string;
    accountAvatarUrl: string;
};

export const ProjectMemberRole = {
    Owner: "Owner",
    Manager: "Manager",
    Member: "Member"
} as const;

export type ProjectMemberRole = typeof ProjectMemberRole[keyof typeof ProjectMemberRole];

export const getProjectMeberRole = (role: ProjectMemberRole): string => {
    const roles: Record<ProjectMemberRole, string> = {
        "Owner": "Proje Sahibi",
        "Manager": "Proje Yöneticisi",
        "Member": "Proje Üyesi"
    };
    return roles[role];
};

export const getAllProjectMemberRoles = (): ProjectMemberRole[] => {
    return Object.values(ProjectMemberRole);
}